### ISSUE TITLE
Can't open multiple YouTube videos via Safari extension

### ISSUE THREAD

**Issue by JohnClay** [iina/iina#2472]
As of 1.0.4, the Safari extension will only open a single video/stream. If you use the extension on a different page, it will replace the previously opened stream.

If you open the URL via the "Open URL in new window" option, that works as expected. The Safari extension now plays the video/stream over THAT window.

**System and IINA version:**

- macOS 10.14.5
- IINA 1.0.4 (107)

**Expected behavior:**

Each video/stream should open in a new window.

**Actual behavior:**

Each video/stream is forced into the same window as the previous file, interrupting playback.

**Steps to reproduce:**

Open video/stream via the Safari extension.

Open a second video/stream via the Safari extension.

All the time.

**Comment by orion1vi** [comment]
`let url = URL(string: "iina://weblink?url=\(escapedURL)")` 
should be changed to 
`let url = URL(string: "iina://weblink?url=\(escapedURL)&new_window=1")`
in `SafariExtensionHandler.swift`, then it would work as expected.

**Comment by anohren** [comment]
Just to be clear, this happens even when you have "always open media in new window" enabled?

**Comment by JohnClay** [comment]
> Just to be clear, this happens even when you have "always open media in new window" enabled?

Yes, that's correct. It's a new behaviour in 1.0.4 w/o any change in setting. Dropping back to 1.0.3 fixes it.

**Comment by orion1vi** [comment]
> "always open media in new window" enabled
When parsing urls, user defaults aren't checked, only the arguments that are parsed together. 
Personally, I don't see the use case of opening videos over existing stream from Safari, it seems that it would only cause annoyance, e.g. music playlist is playing in iina and then video opens over it.

**Comment by anohren** [comment]
Regardless of its actual behavior, that setting is worded in no uncertain terms. It should probably either be honored or rephrased. AFAIR the menu options actually change into "Open URL in new window..." depending on that setting, so expecting the same from the browser doesn't feel far fetched.

Reusing a music mode window for videos doesn't sound like a good strategy, but on the other hand earmarking a window as most recently used by a browser extension and only then reuse it could save the annoyance of having 15 simultaneous video windows open by the browser.

**Comment by JohnClay** [comment]
This is still an issue in 1.0.6 (113).

**Comment by anohren** [comment]
Maybe because the extension technically has the option to “open in new iina window”

**Comment by JohnClay** [comment]
> Maybe because the extension technically has the option to “open in new iina window”

How? Clicking the extension's toolbar button or choosing Open in IINA from the contextual menu provides no options and simply opens in the top window in IINA.

**Comment by anohren** [comment]
At least the Firefox version does, but it has to be enabled in the extension’s settings first and only works from the toolbar there. I added those options to the context menu myself though because I use them a lot.

**Comment by JohnClay** [comment]
> At least the Firefox version does, but it has to be enabled in the extension’s settings first and only works from the toolbar there. I added those options to the context menu myself though because I use them a lot.

Hm, interesting. Definitely not an option in Safari.

**Comment by anohren** [comment]
Yeah that doesn’t sound ideal. It should be capable of it. 

I’ve sort of given up on desktop safari myself. It can’t even sync tabs with my phone nowadays, and Dark Reader that creates dark themes for every webpage on the fly in Firefox is hard to live without once I got used to it.

**Comment by JohnClay** [comment]
Hopefully we can get #2483 merged, if it actually fixes this issue in Safari.
